airbyte_cdk/entrypoint.py

```diff
@@ -141,19 +141,35 @@ def run(self, parsed_args: argparse.Namespace) -> Iterable[str]:
                 )
                 if cmd == "spec":
                     message = AirbyteMessage(type=Type.SPEC, spec=source_spec)
-                    yield from [
+                    yield from (
                         self.airbyte_message_to_string(queued_message)
                         for queued_message in self._emit_queued_messages(self.source)
-                    ]
+                    )
                     yield self.airbyte_message_to_string(message)
+                elif (
+                    cmd == "discover"
+                    and not parsed_args.config
+                    and hasattr(self.source, "check_config_during_discover")
+                    and self.source.check_config_during_discover
```

Comment:
@aaronsteers Am I reading this right that if `check_config_during_discover` is true then we let you have an empty config?

Shouldn't it be in reverse?